packages/access-client/src/agent-use-cases.js

```diff
@@ -170,6 +185,34 @@ export async function waitForDelegationOnSocket(access, opts) {
   throw new TypeError('Failed to get delegation')
 }
 
+/**
+ * @typedef {{signal?: AbortSignal }} AuthorizationWaiterOpts
+ * @typedef {(accessAgent: AccessAgent, opts: AuthorizationWaiterOpts) => Promise<Iterable<Ucanto.Delegation>> } AuthorizationWaiter
+ */
+
+/**
+ * @type AuthorizationWaiter
```

Comment:
love having the type assertion in here to automate ensuring which functions are expected to be drop in replacements to one another

Reply:
a line of doc or two on this might helpfully show up in intellisense for the 2+ functions that impl it

Reply:
great call! done